x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -430,6 +430,7 @@ private boolean isAllowAll() {
 
         private void setAllowAll(boolean allowAll) {
             this.allowAll = allowAll;
+            // TODO: Should we clear queries, i.e. queries = null?
```

Comment:
Can you expand?

Reply:
It is not important anymore and I will remove the comment. But I'll explain myself:

I initially wrote down the comment because the `addAll` method in the same class does not add given queries if `allowAll` is `true`, i.e.:
https://github.com/elastic/elasticsearch/blob/9c6a5fd09bf2af5c17000f995ad2ef2048c4c280/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java#L426-L433

So the code is asymmetric between the two methods `addAll` and `setAllowAll`. For example, if `setAllowAll` is called first, `addAll` becomes no-op. But if `addAll` is called first, `setAllowAll` still works but the class internally still keeps the queries set by the previous call to `addAll`. 

Initially I thought this could be a subtle bug. But it is not because this class is used just as an intermediate variable and the final results are built by checking `isAllowAll` first. This means, if `isAllowAll() == true`, the queries are skipped anyway regardless whether they are empty or not:
https://github.com/elastic/elasticsearch/blob/9c6a5fd09bf2af5c17000f995ad2ef2048c4c280/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java#L284-L285

So overall, there was no issue. We could still make the two methods more symmetric, but it doesn't really help solving any problem.